client/src/lib/utils/Application/createOpportunity.ts

```diff
@@ -0,0 +1,48 @@
+// import { v4 as uuidv4 } from "uuid";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Database } from "../../types/database.types";
+
+
+/**
+ * Initializes an oportunity for a opportunity table, if it is not exist.
+ *
+ * @param opportunity - The ID of the opportunity as uuid.
+ * @param roleTitle - The role title of the opportunity.
+ * @param companyName -  The company name of the opportunity.
+ * @param type - The type of the opportunity.
+ * @returns A promise that resolves to opportunity or null if an error occurs or no thread is found.
+    type: OpportunityType,
+*/
+
+export async function createOpportunity(
+    supabase: SupabaseClient,
```

Comment:
make it an optional and be the last parameter of the function so it can be accessed by other components if need be: supabase = createSupabaseClient()

Reply:
Fixed